cpp_generator.py

```diff
@@ -0,0 +1,367 @@
+        
+        return "\n\n".join(collection_types)
+
+    def __gen_model(self, node: VSSNode, path: str, is_root=False):
+        # must be done before generating the imports, since it is adding imports to the list
+        collection_types = self.__gen_collection_types(node, path)
+
+        self.__gen_header(node)
+        self.__gen_imports(node)
+        self.ctx_header.write(self.__generate_opening_namespace_text("velocitas", node))
+        self.ctx_header.write("using ParentClass = Model;\n\n")
+        self.__gen_model_docstring(node)
+        self.ctx_header.write(f"class {node.name} : public ParentClass {{\n")
+        self.ctx_header.write("public:\n")
+
+        with self.ctx_header as header_public, self.ctx_source as source_public:
+            header_public.write(collection_types)
+            header_public.write("\n\n")
+
+            if is_root:
+                header_public.write(f"{node.name}();\n")
+            else:
+                header_public.write(f"{node.name}(const std::string& name, ParentClass* parent);\n")
+
+            if node.children:
+                header_public.write("\n")
+
+            if is_root:
+                source_public.write(f"{node.name}::{node.name}() :\n")
+                source_public.indent()
+                source_public.write("ParentClass(\"Vehicle\")")
+            else:
+                source_public.write(f"{node.name}::{node.name}(const std::string& name, ParentClass* parent) :\n")
+                source_public.indent() 
+                source_public.write("ParentClass(name, parent)")
+
+            # create members
+            for child in node.children:
+                child_namespace = "::".join(self.__get_namespace_for_node(child))
+                self.__document_member(child)
+                
+                if child.type.value in ("attribute", "sensor", "actuator"):
+                    header_public.write(f"DataPoint{self.__get_data_type(child.datatype.value)} {child.name};\n\n")
+                    source_public.write(",\n" + f"{child.name}(\"{child.name}\", this)")
+                
+                if child.type == VSSType.BRANCH:
+                    if child.instances:
+                        header_public.write(
+                            f"{child.name}Collection {child.name};\n"
+                        )
+                        source_public.write(",\n" + f"{child.name}(this)")
+                    else:
+                        header_public.write(f"{child_namespace}::{child.name} {child.name};\n\n")
+                        source_public.write(",\n" + f"{child.name}(\"{child.name}\", this)")
+
+        self.ctx_header.write("};\n")
+        self.ctx_source.write("\n{}\n")
+
+        self.ctx_source.dedent()
+
+        self.__gen_footer(node)
+
+        with open(os.path.join(path, f"{node.name}.hpp"), "w", encoding="utf-8") as file:
```

Comment:
The header files should ideally be output to a `<root>/include` path while the source files should be output to `<root>/src`